src/utils.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright 2020 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use error_chain::bail;
+use std::process::Command;
+
+use crate::errors::*;
+
+#[macro_export]
+macro_rules! runcmd {
+    ($cmd:expr, $($args:expr),*) => {{
+        let mut cmd = Command::new($cmd);
+        $( cmd.arg($args); )*
+        // print full cmdline tried here?
+        cmd.status().chain_err(|| format!("running `{}`", $cmd))
+    }}
+}
+
+#[macro_export]
+macro_rules! runcmd_output {
+    ($cmd:expr, $($args:expr),*) => {{
+        let mut cmd = Command::new($cmd);
+        $( cmd.arg($args); )*
+        // print full cmdline tried here?
+        cmd_output(&mut cmd).chain_err(|| format!("running `{}`", $cmd))
+    }}
+}
+
+pub fn cmd_output(cmd: &mut Command) -> Result<String> {
+    let result = cmd.output().chain_err(|| "executing")?;
+    if !result.status.success() {
+        bail!("{}", String::from_utf8_lossy(&result.stderr));
```

Comment:
The failure isn't that the program produced stderr, but that it exited nonzero.  We swallow stderr except on failure because some of the programs we call are noisy.

```suggestion
        eprint!("{}", String::from_utf8_lossy(&result.stderr));
        bail!("{:#?} failed: {}", cmd, result.status);
```

Reply:
I just don't like a utility function printing to stderr like this vs just making it part of the error message. IOW, I think I'd prefer

```rust
bail!("{:#?} failed with {}; stderr: {}", cmd, result.status, String::from_utf8_lossy(&result.stderr));
```

Thoughts?

Reply:
In general I agree.  In this case, I don't think of it as "printing to stderr" so much as "conditionally swallowing stderr".  We're just forwarding what the program originally sent there, which might be many lines.

Reply:
OK, adapted. Though in that spirit, I don't think we should decode as UTF-8 at all. We should just dump it as is to `stderr`. Will leave that as a follow-up.